platform/macos/display_server_macos.mm

```diff
@@ -3797,18 +3797,33 @@
 
 	if (rendering_context) {
 		if (rendering_context->initialize() != OK) {
-			memdelete(rendering_context);
-			rendering_context = nullptr;
+			bool failed = true;
```

Comment:
Same issue here, removing this results in a memory leak if Vulkan is enabled and `fallback_to_vulkan` is `true`.